src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -980,27 +1074,69 @@ private async Task<Dictionary<KeyValueIdentifier, ConfigurationSetting>> LoadKey
 
                 KeyValueIdentifier watchedKeyLabel = new KeyValueIdentifier(watchedKey, watchedLabel);
 
-                // Skip the loading for the key-value in case it has already been loaded
-                if (existingSettings.TryGetValue(watchedKey, out ConfigurationSetting loadedKv)
-                    && watchedKeyLabel.Equals(new KeyValueIdentifier(loadedKv.Key, loadedKv.Label)))
-                {
-                    watchedIndividualKvs[watchedKeyLabel] = new ConfigurationSetting(loadedKv.Key, loadedKv.Value, loadedKv.Label, loadedKv.ETag);
-                    continue;
-                }
-
                 // Send a request to retrieve key-value since it may be either not loaded or loaded with a different label or different casing
                 ConfigurationSetting watchedKv = null;
                 try
                 {
-                    await CallWithRequestTracing(async () => watchedKv = await client.GetConfigurationSettingAsync(watchedKey, watchedLabel, cancellationToken).ConfigureAwait(false)).ConfigureAwait(false);
+                    Response<ConfigurationSetting> response = null;
+                    await CallWithRequestTracing(async () =>
+                    {
+                        response = await client.GetConfigurationSettingAsync(watchedKey, watchedLabel, cancellationToken).ConfigureAwait(false);
+                    }).ConfigureAwait(false);
+
+                    using Response rawResponse = response.GetRawResponse();
+                    DateTimeOffset responseDate = rawResponse.GetDate();
+                    if (_watchedIndividualKvChangeDetectedTime.TryGetValue(watchedKeyLabel, out DateTimeOffset lastChangeDetectedTime))
```

Comment:
Date comparison shouldn't be necessary for loading. Only for detecting if a refresh/reload is needed.